packages/app/src/app.ts

```diff
@@ -387,17 +391,30 @@ export class App<Req extends Request = Request, Res extends Response = Response>
 
     let idx = 0
 
-    const loop = () => res.writableEnded || (idx < mw.length && handle(mw[idx++])(req, res, next))
+    const handleNext = (): void => void handle(mw[idx++])(req, res, next)
 
-    next = next || ((err) => (err ? this.onError(err, req, res) : loop()))
+    const parentNext = next
+    next = (err) => {
+      if (err != null) {
+        return this.onError(err, req, res)
+      }
+
+      if (res.writableEnded) return
+      if (idx >= mw.length) {
+        if (parentNext != null) parentNext()
+        return
+      }
+
+      handleNext()
+    }
 
-    loop()
+    handleNext()
   }
 
   /**
    * Creates HTTP server and dispatches middleware
    * @param port server listening port
-   * @param Server callback after server starts listening
```

Comment:
unrelated change

Reply:
My IDE was giving me grief about the jsdoc parameter being missing. Moving this change (and the other jsdoc one above) to another PR would be a waste of my time.

Reply:
ok